config/index.ts

```diff
@@ -1,282 +1,426 @@
+        logger.debug(
+          i18n('config.isConfigValid.invalidAccountName', {
+            accountName: account.name,
+          })
+        );
+        return false;
+      }
+      accountNamesMap[account.name] = true;
+    }
+
+    accountIdsMap[account.accountId] = true;
+    return true;
+  });
 }
 
-export function getConfigPath(
-  path?: string,
-  useHiddenConfig = false
-): string | null {
-  if (useHiddenConfig || CLIConfiguration.isActive()) {
-    return getConfigFilePath();
-  }
-  return config_DEPRECATED.getConfigPath(path);
+export function createEmptyConfigFile(useGlobalConfig = false): void {
+  const { configFilePathFromEnvironment } = getConfigPathEnvironmentVariables();
+  const defaultPath = useGlobalConfig
+    ? getGlobalConfigFilePath()
+    : getLocalConfigDefaultFilePath();
+
+  const pathToWrite = configFilePathFromEnvironment || defaultPath;
+
+  writeConfigFile({ accounts: [] }, pathToWrite);
 }
 
-export function configFileExists(useHiddenConfig?: boolean) {
-  return useHiddenConfig
-    ? newConfigFileExists()
-    : Boolean(config_DEPRECATED.getConfigPath());
+export function deleteConfigFile(): void {
+  const pathToDelete = getConfigFilePath();
+  fs.unlinkSync(pathToDelete);
 }
 
-export function getAccountConfig(accountId?: number): CLIAccount | null {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.getConfigForAccount(accountId);
+export function getConfigAccountById(accountId: number): HubSpotConfigAccount {
+  const { accounts } = getConfig();
+
+  const account = getConfigAccountByIdentifier(
+    accounts,
+    ACCOUNT_IDENTIFIERS.ACCOUNT_ID,
+    accountId
+  );
+
+  if (!account) {
+    throw new Error(i18n('config.getConfigAccountById.error', { accountId }));
   }
-  return config_DEPRECATED.getAccountConfig(accountId) || null;
+
+  return account;
 }
 
-export function accountNameExistsInConfig(name: string): boolean {
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.isAccountInConfig(name);
+export function getConfigAccountByName(
+  accountName: string
+): HubSpotConfigAccount {
+  const { accounts } = getConfig();
+
+  const account = getConfigAccountByIdentifier(
+    accounts,
+    ACCOUNT_IDENTIFIERS.NAME,
+    accountName
+  );
+
+  if (!account) {
+    throw new Error(
+      i18n('config.getConfigAccountByName.error', { accountName })
+    );
   }
-  return config_DEPRECATED.accountNameExistsInConfig(name);
+
+  return account;
 }
 
-export function updateAccountConfig(
-  configOptions: Partial<FlatAccountFields>
-): FlatAccountFields | null {
-  const accountIdentifier = getAccountIdentifier(configOptions);
-  if (CLIConfiguration.isActive()) {
-    return CLIConfiguration.addOrUpdateAccount({
-      ...configOptions,
-      accountId: accountIdentifier,
-    });
-  }
-  return config_DEPRECATED.updateAccountConfig({
-    ...configOptions,
-    portalId: accountIdentifier,
-  });
+export function getConfigAccountIfExists(
```

Comment:
I was reviewing one of your PR's in the CLI, and this util tripped me up a little. This is effectively the same functionality as `getConfigAccountByInferredIdentifier`. But the naming of this util doesn't make it clear that we're inferring the identifier.`

[Also here's a relevant comment from a CLI PR](https://github.com/HubSpot/hubspot-cli/pull/1392/files#r2002008951)